inst/include/sampler.h

```diff
@@ -403,7 +413,71 @@ namespace smc {
+        int historylength = static_cast<int>(lHistorylength);
+        /// Output container for particle population ancestor lines
+        std::vector<population<Space> > populationAL(historylength);
+        /// Get final iteration indices
+        arma::Col<unsigned int> AI_tmp = History[historylength-1].GetAindices();
+        /// Get final iteration particle values
+        populationAL[historylength-1] = GetHistoryPopulation(historylength-1);
+        /// Iterating from last iteration to initiliaization period whereby:
+        /// 1. Re-ordering particle population elements for each t=0,...,T-1 accoridng to ancestral lines
+        /// 2. Storing these particle population classes in output: populationAL
+        if (historylength > 1){
+            for (int t = historylength - 2; t >= 0; --t){
+                for(int i = 0; i < static_cast<int>(N); ++i){
+                    /// Setting each particle population element at indices i=0,...,N-1 to its ancesor index in the output vector populationAL
+                    populationAL[t].SetValueN(GetHistoryPopulation(t).GetValueN(static_cast<int>(AI_tmp(i))), i);
+                }
+                /// Re-mapping ancestor indices to retrieve ancestor indices a period before
+                AI_tmp = (History[t].GetAindices()).elem(AI_tmp);
+            }
+        }
+        /// The output container stores population classes re-ordered according to their ancestor lines for t=0,...,T-1. So, for example, the first each element of populationAL[t] is the direct ancestor population class element of populationAL[t+1].
+        return populationAL;
+    }
+    template <class Space, class Params>
+    arma::Col<unsigned int> sampler<Space, Params>::GetALineInd(long n) const
+    // void sampler<Space, Params>::GetALineInd(long n) const
+    {
+        long N = GetNumber();
+        long lHistorylength = GetHistoryLength();
+        int historylength = static_cast<int>(lHistorylength);
+
+        /// Get final iteration indices and final iteration particle values
+        unsigned int AI_tmp = (History[historylength-1].GetAindices()).at(static_cast<int>(n));
+        arma::Col<unsigned int> AlineInd(historylength);
+        AlineInd[historylength - 1] = static_cast<int>(n);
+
+        for(int t = historylength - 2; t >= 0; --t){
+            AI_tmp = (History[t].GetAindices()).at(AI_tmp);
+            AlineInd[t] = AI_tmp;
+        }
+        return AlineInd;
+    }
+    template <class Space, class Params>
+    std::vector<Space> sampler<Space,Params>::GetALineSpace(long n) const
+    {
+        long N = GetNumber();
+        long lHistorylength = GetHistoryLength();
+        int historylength = static_cast<int>(lHistorylength);
+
+        std::vector<Space> AlineSpace(historylength);
+
+        /// Get final iteration indices and final iteration particle values
+        arma::Col<unsigned int> AlineInd(historylength);
+        AlineInd = this->GetALineInd(static_cast<int>(n));

```

Comment:
Again there seems to be some inconsistency around case with these As. I don't want to obsess over this, but it does jar a bit when the function is `GetALineInd` and the variable is `AlineInd`.

Reply:
Done, will keep in mind for next time!